browser/ui/browser_commands.cc

```diff
@@ -81,17 +82,36 @@ void OpenGuestProfile() {
   profiles::SwitchToGuestProfile(ProfileManager::CreateCallback());
 }
 
-void ToggleSpeedreader(Browser* browser) {
+void MaybeDistillAndShowSpeedreaderBubble(Browser* browser) {
 #if BUILDFLAG(ENABLE_SPEEDREADER)
   speedreader::SpeedreaderService* service =
       speedreader::SpeedreaderServiceFactory::GetForProfile(browser->profile());
   if (service) {
-    // This will trigger a button update via a pref change subscribition.
-    service->ToggleSpeedreader();
-
     WebContents* contents = browser->tab_strip_model()->GetActiveWebContents();
     if (contents) {
-      contents->GetController().Reload(content::ReloadType::NORMAL, false);
+      auto* tab_helper =
+          speedreader::SpeedreaderTabHelper::FromWebContents(contents);
+      if (!tab_helper)
+        return;
+
+      const bool speedreader_enabled = tab_helper->IsSpeedreaderEnabled();
+      const DistillState state = tab_helper->PageDistillState();
+
+      if (state == DistillState::kSpeedreaderMode) {
+        tab_helper->ShowSpeedreaderBubble();
+      } else if (state == DistillState::kReaderMode) {
+        tab_helper->ShowReaderModeBubble();
+      } else {
+        // If this is called on an undistilled page, we single shot it.
+        tab_helper->SingleShotSpeedreader();
+        if (speedreader_enabled && !tab_helper->IsEnabledForSite()) {
+          // Speedreader is enabled. Allow the user to remove this site from the
+          // blacklist.
+          tab_helper->ShowSpeedreaderBubble();
+        } else {
+          tab_helper->ShowReaderModeBubble();
```

Comment:
I would add a comment saying how can we get there

Reply:
added a bunch of comments about these different states